### PR TITLE
add page to upload files

### PR DESCRIPTION
**Pull request by lordlinus**
## Purpose

Add page to upload PDF files directly from the UI instead of running separate script

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
azd up
cd app
run start.cmd or start.sh
go to http://127.0.0.1:5000/#/ul
upload the PDF files from the repository data directory
```

* Test the code

Manual test
```
check console for success of file upload and index status
```

## What to Check
Verify that the following are valid:
* check the chat and  ask your data is working as expected

## Other Information

UI page could make use of status message better (need help on react code to update progress for long running jobs)

### CODE REVIEW

File: infra/main.bicep [hunk 1]
```diff
@@ -190,7 +190,7 @@ module openAiRoleUser 'core/security/role.bicep' = {
   params: {
     principalId: principalId
     roleDefinitionId: '5e0bd9bd-7b93-4f28-af87-19fc36ad61bd'
-    principalType: 'User'
+    principalType: 'ServicePrincipal'
```
Comment (jongio) [inline infra/main.bicep:193]: These roles are for the local user, please do not change to ServicePrincipal

Comment (lordlinus) [reply infra/main.bicep]: done

File: .github/workflows/uploadfiles_app-backend-zkrnwcgqiomq6.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
+# More GitHub Actions for Azure: https://github.com/Azure/actions
+# More info on Python, GitHub Actions, and Azure App Service: https://aka.ms/python-webapps-actions
+
+name: Build and deploy Python app to Azure Web App - app-backend-zkrnwcgqiomq6
```
Comment (jongio) [inline .github/workflows/uploadfiles_app-backend-zkrnwcgqiomq6.yml:5]: You can remove most of the steps and call `azd deploy` in one step.  See this workflow for an example: https://github.com/Azure-Samples/todo-python-mongo/blob/main/.github/workflows/azure-dev.yml

Comment (lordlinus) [reply .github/workflows/uploadfiles_app-backend-zkrnwcgqiomq6.yml:5]: nice @jongio , Thanks for sharing, will update

**Comment by itmilos** [comment]
{'error': 'This request is not authorized to perform this operation using this permission.\nRequestId:36f8347b-b01e-0065-4ba2-8dc669000000\nTime:2023-05-23T18:17:38.0661381Z\nErrorCode:AuthorizationPermissionMismatch\nContent: <?xml version="1.0" encoding="utf-8"?><Error><Code>AuthorizationPermissionMismatch</Code><Message>This request is not authorized to perform this operation using this permission.\nRequestId:36f8347b-b01e-0065-4ba2-8dc669000000\nTime:2023-05-23T18:17:38.0661381Z</Message></Error>'}

I get his @lordlinus how do you handle connection to blob storage here ?

**Comment by TimoRiikonen** [comment]
Hello,

Has this branch got stale?
At least I didn't manage to upload files.

Exception has occurred: HttpResponseError
() The request is invalid. Details: parameters : The property 'category' does not exist on type 'search.documentFields'. Make sure to only use property names that are defined by the type.
Code: 
Message: The request is invalid. Details: parameters : The property 'category' does not exist on type 'search.documentFields'. Make sure to only use property names that are defined by the type.
  File "/workspaces/azure-search-openai-demo/app/backend/indexdocs.py", line 151, in index_sections

image

**Comment by TimoRiikonenIF** [comment]
My colleague who has admin permissions got this running.

**Comment by superpoussin22** [comment]
doesn't not work for me , even if the UPLOAD_FOLDER variable  is mapped to a file storage in the webapp.
I can see the file in the storage but the ui always says :
"Some files failed to upload"
@lordlinus  any idea 

thanks in advance

**Comment by TimoRiikonen** [comment]
Yes, I didn't have admin permissions at the time.
I should have had blob upload permissions, but I am not 100% sure.

**Comment by superpoussin22** [comment]
the bicep file need to be updated:
the app identity must have rights on index (search )  , storage and blob to be able to work :)
thoses rights must be added at RG level 
@TimoRiikonen @lordlinus

**Comment by AIconcepts2023** [comment]
Hi, sorry if this is a silly question, but I get an error trying to "azd up" because the "FileUploader.tsx" states on line 6 "Import { uploadApi, UploadFileRequest, UploadFileResponse } from "../../api";

However in app/frontend/src/api/api.ts there are no exports "UploadApi", "UploadFileRequest", "UploadFileResponse". 

Has anyone encountered the same issue please?

Many thanks for your help

**Comment by beouk** [comment]
@pamelafox looking at building my own UI upload solution but wanted to see if there's an estimate for an 'official' feature?

**Comment by pamelafox** [comment]
Per other thread, am working on it now and will aim to send WIP PR today (but it will not have all the components needed).

**Comment by pamelafox** [comment]
Update: Now that #1395 is nearing mergability, I am closing this PR. Thanks so much for the contribution!

**Pull request closed without merging**